include/zephyr/drivers/gpio.h

```diff
@@ -456,6 +457,7 @@ struct gpio_driver_config {
  * element in the driver's struct driver_data declaration.
  */
 struct gpio_driver_data {
+	DEVICE_MMIO_RAM; /* Must be first */
```

Comment:
NAK on these changes as they will increase the ROM/RAM footprint needlessly for all other GPIO drivers. Please seek inspiration in `drivers/gpio/gpio_intel.c` on how to contain MMIO within the GPIO driver itself.

Reply:
done